pymc_marketing/clv/utils.py

```diff
@@ -42,42 +42,37 @@ def to_xarray(customer_id, *arrays, dim: str = "customer_id"):
 
 def customer_lifetime_value(
     transaction_model,
-    customer_id: pd.Series | np.ndarray,
-    frequency: pd.Series | np.ndarray,
-    recency: pd.Series | np.ndarray,
-    T: pd.Series | np.ndarray,
+    transaction_data: pd.DataFrame,
     monetary_value: pd.Series | np.ndarray | xarray.DataArray,
-    time: int = 12,
-    discount_rate: float = 0.01,
-    freq: str = "D",
+    future_t: int = 12,
+    discount_rate: float = 0.00,
+    time_unit: str = "D",
 ) -> xarray.DataArray:
     """
-    Compute the average lifetime value for a group of one or more customers.
-    This method computes the average lifetime value for a group of one or more customers.
-    It also applies Discounted Cash Flow.
+    Compute the average lifetime value for a group of one or more customers,
+    and apply a discount rate for net present value estimations.
+    Note `future_t` is measured in months regardless of `time_unit` specified.
 
     Adapted from lifetimes package
     https://github.com/CamDavidsonPilon/lifetimes/blob/41e394923ad72b17b5da93e88cfabab43f51abe2/lifetimes/utils.py#L449
 
     Parameters
     ----------
     transaction_model: CLVModel
-        The model to predict future transactions
-    customer_id: array_like
-        Customer unique identifiers. Must not repeat.
-    frequency: array_like
-        The frequency vector of customers' purchases (denoted x in literature).
-    recency: array_like
-        The recency vector of customers' purchases (denoted t_x in literature).
-    T: array_like
-        The vector of customers' age (time since first purchase)
+        Predictive model for future transactions. BG/NBD and Pareto/NBD are currently supported.
+    transaction_data: pandas.DataFrame
```

Comment:
```suggestion
    transaction_data: ~pandas.DataFrame
```

Reply:
That's a good point about the `xarray` broadcasting support. I'm not quite sure what that will look like yet, so for now I'll combine these parameters.

I've added a check for a "future_spend" column to highlight the fact this utility function is expecting predicted values (the `GammaGammaModel` method runs predictions prior to calling this utility). I also had to make a small change to one of the tests because assigning predictions to a dataframe column is creating tiny rounding errors on the order of `1e-15`.